src/util/TokenSpawnTool.ttslua

```diff
@@ -74,6 +86,21 @@ function onScriptingButtonDown(index, playerColor)
         end
       end
     end
+  -- check for nearest investigator card and change action token state to its class 
+  elseif tokenType == "actionToken" then
+    local matColor = playmatApi.getMatColorByPosition(position)
+    local result = playmatApi.searchAroundPlaymat(matColor, "isCard")
+    for j, card in ipairs(result) do
```

Comment:
Should add `local investigatorClass = "Neutral"` before the loop to
a) not define it globally
b) cover the case of not finding an investigator card (e.g. because metadata is missing)